### PR TITLE
packagekit: 1.1.12 -> 1.1.13

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://github.com/hughsie/PackageKit/blob/PACKAGEKIT_1_1_13/NEWS

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dtzWill** [comment]
@GrahamcOfBorg test packagekit

**Comment by worldofpeace** [comment]
I should probably add myself as a maintainer to this, I can at least receive pings on it.

**Review by nh2 — APPROVED** [review]
`nix-review` says:

```
[14 built, 741 copied (2556.1 MiB), 505.7 MiB DL]
https://github.com/NixOS/nixpkgs/pull/77498
8 package built:
discover gnome3.gnome-packagekit gnome3.gnome-software simple-scan packagekit packagekit-qt pantheon.appcenter system-config-printer
```

**Comment by worldofpeace** [comment]
This looks interesting from the changelog
```
 - Add support for user services (Julian Andres Klode)
```

I believe that is this commit https://github.com/hughsie/PackageKit/commit/d9fe72020cb51179cefccf93e33f9cdab9ad0b37.
The configure output had this:
```
systemd system unit dir:   /nix/store/1jjkn934bfhma9yayp3pvzdpznjj87lk-packagekit-1.1.13/lib/systemd/system/
systemd user unit dir:     /nix/store/a23ym95iyzhszm4a6yj83dfcfsvjbdzj-systemd-243.3/lib/systemd/user
```

I notice in the build output with this PR there's a system unit installed but no user services.
So I suspect because the directory for the user unit dir is writable it failed silently?
We should probably set `--with-systemduserunitdir`. 
A better solution might be for them to use `--define-variable` in
```
[has_systemduserunitdir=$($PKG_CONFIG --variable=systemduserunitdir systemd)])
```

**Comment by worldofpeace** [comment]
Ooh they jst switched to meson :sparkles: 
* https://github.com/hughsie/PackageKit/commit/f144fbc20df1aa21e666ef2256adbbc63ffcf131

**Review by worldofpeace — CHANGES_REQUESTED** [review]
See https://github.com/NixOS/nixpkgs/pull/77498#issuecomment-573449167.

**Comment by dtzWill** [comment]
> Ooh they just switched to meson sparkles

:sparkles:

(for folks following along, FYI this was done after the release of this version, so meson is for "next time" :smile_cat: )

> This looks interesting from the changelog
> 
> ```
>  - Add support for user services (Julian Andres Klode)
> ```
> 
> I believe that is this commit [hughsie/PackageKit@d9fe720](https://github.com/hughsie/PackageKit/commit/d9fe72020cb51179cefccf93e33f9cdab9ad0b37).
> The configure output had this:
> 
> ```
> systemd system unit dir:   /nix/store/1jjkn934bfhma9yayp3pvzdpznjj87lk-packagekit-1.1.13/lib/systemd/system/
> systemd user unit dir:     /nix/store/a23ym95iyzhszm4a6yj83dfcfsvjbdzj-systemd-243.3/lib/systemd/user
> ```

Eep, good catch. Thanks!

> 
> I notice in the build output with this PR there's a system unit installed but no user services.
> So I suspect because the directory for the user unit dir is writable it failed silently?

This doesn't appear to be the case, as nothing is installed when setting the directory somewhere it can write (as suggested below and in most recent commit).

> We should probably set `--with-systemduserunitdir`.

Pushed commit doing this!

> A better solution might be for them to use `--define-variable` in

Maybe submit an upstream PR? *friendly nudge* (assuming that seems like a good idea, couldn't say)

-----

Wasn't sure if we should point it to `$out/lib/systemd` or `$out/etc/systemd` or indeed perhaps elsewhere (so it can "discover" user services, which maybe is what it is trying for? (??)).  If it matters, or there are guidelines for this, I'm happy to be educated or pointed to where such is documented.  grep'ing for similar arguments elsewhere in-tree shows a mixed batch, so wasn't sure.  Thanks :).

File: pkgs/tools/package-management/packagekit/default.nix [hunk 1]
```diff
@@ -39,8 +39,9 @@ stdenv.mkDerivation rec {
     "--disable-offline-update"
     "--localstatedir=/var"
     "--sysconfdir=/etc"
-    "--with-dbus-sys=${placeholder "out"}/share/dbus-1/system.d"
-    "--with-systemdsystemunitdir=${placeholder "out"}/lib/systemd/system/"
+    "--with-dbus-sys=${placeholder "out"}/etc/dbus-1/system.d"
```
Comment (worldofpeace) [inline pkgs/tools/package-management/packagekit/default.nix]:
```suggestion
    "--with-dbus-sys=${placeholder "out"}/share/dbus-1/system.d"
```

the other dir is deprecated

**Comment by worldofpeace** [comment]
@GrahamcOfBorg test packagekit

**Comment by nh2** [comment]
@worldofpeace Is green, good to go or more to do?

**Comment by worldofpeace** [comment]
> @worldofpeace Is green, good to go or more to do?

I believe so, though the changelog was slightly confusing.

**Pull request merged**